### PR TITLE
Clean up module documentation

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
This is a backport of #36909

This PR includes:
- Removal of `default: None` (and variations)
- Removal of `required: false`
- Fixing booleans and `type: bool` where required

This PR also fixed 673 validation issues from
*test/sanity/validate-modules/ignore.txt*

This relates to #36901

##### ISSUE TYPE
 - Feature Pull Request
 - Docs Pull Request

##### COMPONENT NAME
module docs

##### ANSIBLE VERSION
v2.5

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/ec2_asg.py [hunk 1]
```diff
@@ -169,10 +148,33 @@
   suspend_processes:
     description:
       - A list of scaling processes to suspend.
-    required: False
     default: []
     choices: ['Launch', 'Terminate', 'HealthCheck', 'ReplaceUnhealthy', 'AZRebalance', 'AlarmNotification', 'ScheduledActions', 'AddToLoadBalancer']
     version_added: "2.3"
+  metrics_collection:
```
Comment (dagwieers) [inline lib/ansible/modules/cloud/amazon/ec2_asg.py]: Ok, so it is weird that we're backporting modules parameter improvements in v2.5 ?
Shouldn't this already be in v2.5 ??

**Comment by dagwieers** [comment]
@nitzmahone @gundalow If we want to do this, we shouldn't wait to long before merging.

**Comment by gundalow** [comment]
What's the plan for this?

**Comment by dagwieers** [comment]
Hmm, I thought this was already merged in v2.5. What do you think we should do now ?

**Comment by gundalow** [comment]
Not going to have time to rebase, review and merge this before we cut `stable-2.6` branch, so lets close this and we can review during 2.7 development.

**Pull request closed without merging**